### PR TITLE
Read term_common_params from the state

### PR DESCRIPTION
**Pull request by Remagpie**
Copied from https://github.com/CodeChain-io/codechain/pull/1895
Some parts are revised to match the foundry's situation.

### CODE REVIEW

File: state/src/item/metadata.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub struct Metadata {
     term: TermMetadata,
     seq: u64,
     params: Option<CommonParams>,
+    term_params: Option<CommonParams>,
```
Comment (sgkim126) [inline state/src/item/metadata.rs]: What's the difference between `params` and `term_params`?

Comment (Remagpie) [reply state/src/item/metadata.rs]: `term_params` is the common parameter at the start of the term.
Some parts in our implementation (e.g. tendermint) depend on this value.

File: state/src/impls/top_level.rs [hunk 2]
```diff
@@ -999,6 +999,12 @@ impl TopState for TopLevelState {
         metadata.increase_seq();
         Ok(())
     }
+
+    fn snapshot_term_params(&mut self) -> StateResult<()> {
```
Comment (sgkim126) [inline state/src/impls/top_level.rs]: I don't think snapshot is appropriate for this function. How about renaming it to `update_term_params`?

Comment (Remagpie) [reply state/src/impls/top_level.rs]: Agreed. It was not my decision to name this function like this, but I think your suggestion is better.

**Pull request merged**